src/gpuhunt/providers/runpod.py

```diff
@@ -0,0 +1,273 @@
+import logging
+from concurrent.futures import ThreadPoolExecutor, as_completed
+from itertools import chain
+from typing import List, Optional
+
+import requests
+from requests import RequestException
+
+from gpuhunt._internal.models import QueryFilter, RawCatalogItem
+from gpuhunt.providers import AbstractProvider
+
+logger = logging.getLogger(__name__)
+API_URL = "https://api.runpod.io/graphql"
+
+
+class RunpodProvider(AbstractProvider):
+    NAME = "runpod"
+
+    def get(
+        self, query_filter: Optional[QueryFilter] = None, balance_resources: bool = True
+    ) -> List[RawCatalogItem]:
+        offers = self.fetch_offers()
+        return sorted(offers, key=lambda i: i.price)
+
+    def fetch_offers(self) -> List[RawCatalogItem]:
+        offers = [get_raw_catalog(pod_type) for pod_type in self.list_pods()]
+        return list(chain.from_iterable(offers))
+
+    @staticmethod
+    def list_pods() -> List[dict]:
+        payload_gpu_types = {"query": gpu_types_query, "variables": {}}
+        try:
+            gpu_types = make_request(payload_gpu_types)
+        except RequestException as e:
+            logger.exception("Failed to make request for GPU types: %s", e)
```

Comment:
I suggest to re-raise an RequestException. It will be more understandablу than `NameError(gpu_types)`.
```suggestion
        except RequestException as e:
            logger.exception("Failed to make request for GPU types: %s", e)
            raise
```